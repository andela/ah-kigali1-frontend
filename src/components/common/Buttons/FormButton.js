import React from "react";
import PropTypes from "prop-types";
const FormButton = props => {
  return (
    <input
      {...props}
      type="button"
      value={props.value}
      onClick={props.onClick}
      disabled={props.disabled}
    />
  );
};

FormButton.prototype = {
  value: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};
FormButton.defaultProps = {
  value: "Submit",
  disabled: false
};
export default FormButton;
