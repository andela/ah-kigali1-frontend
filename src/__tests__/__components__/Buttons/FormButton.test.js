import React from "react";
import { shallow } from "enzyme";
import FormButton from "../../../components/common/Buttons/FormButton";

const setUp = props => shallow(<FormButton {...props} />);
const props = {
  value: "Submit",
  onClick: jest.fn()
};
describe("Basic button", () => {
  let component;
  beforeEach(() => {
    component = setUp(props);
  });
  test("should render button", () => {
    expect(component.props().value).toEqual(props.value);
    expect(component.props().type).toEqual("button");
    component.simulate("click");
    expect(component.props().onClick).toHaveBeenCalledTimes(1);
  });
});
