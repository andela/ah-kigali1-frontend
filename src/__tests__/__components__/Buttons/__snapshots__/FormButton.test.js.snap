// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic button should match the snapshot 1`] = `
<input
  disabled={false}
  onClick={[MockFunction]}
  type="button"
  value="Submit"
/>
`;
